src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Telemetry/ITelemetryReporter.cs

```diff
@@ -8,7 +8,9 @@ namespace Microsoft.AspNetCore.Razor.Telemetry;
 
 public interface ITelemetryReporter
 {
+    IDisposable BeginBlock(string name, Severity severity);
+    IDisposable BeginBlock<T>(string name, Severity severity, T values) where T : IImmutableDictionary<string, object?>;
```

Comment:
Does this need the same treatment to remove the generic and constraint, like `ReportEvent` got?